pkg/codegen/go/gen_program.go

```diff
@@ -584,7 +587,64 @@ func (g *generator) genLocalVariable(w io.Writer, v *hcl2.LocalVariable) {
 		g.Fgenf(w, "%s := %.3v;\n", name, expr)
 
 	}
+}
+
+func (g *generator) genConfigVariable(w io.Writer, v *hcl2.ConfigVariable) {
+	if !g.configCreated {
+		g.Fprintf(w, "%sconfig := config.New(ctx, \"\")\n", g.Indent)
+		g.configCreated = true
+	}
+
+	getType := "Object"
+	switch v.Type() {
+	case model.StringType:
+		getType = ""
+	case model.NumberType:
+		getType = "Float"
+	case model.IntType:
+		getType = "Int"
+	case model.BoolType:
+		getType = "Boolean"
+	}
 
+	getOrRequire := "Get"
+	if v.DefaultValue == nil {
+		getOrRequire = "Require"
+	}
+
+	if v.DefaultValue == nil {
+		g.Fgenf(w, "%[1]s%[2]s := config.%[3]s%[4]s(\"%[2]s\")\n", g.Indent, v.Name(), getOrRequire, getType)
+	} else {
+		expr, temps := g.lowerExpression(v.DefaultValue, v.DefaultValue.Type(), false)
+		g.genTemps(w, temps)
+		switch expr := expr.(type) {
+		case *model.FunctionCallExpression:
+			switch expr.Name {
+			case hcl2.Invoke:
+				g.Fgenf(w, "%s, err := %.3v;\n", v.Name(), expr)
+				g.isErrAssigned = true
+				g.Fgenf(w, "if err != nil {\n")
+				g.Fgenf(w, "return err\n")
+				g.Fgenf(w, "}\n")
+			}
+		default:
+			g.Fgenf(w, "%s := %.3v;\n", v.Name(), expr)
+		}
```

Comment:
This looks pretty similar to `genLocalVariable`, perhaps we should replace this with a call to that function?

Reply:
There are some subtle differences. Chose for the moment to allow a bit of duplication instead of refactoring both functions to share code.